sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -202,6 +205,13 @@ def partition_key(self):
     @partition_key.setter
     def partition_key(self, value):
         # type: (str) -> None
+        if value and len(value) > 128:
+            raise ValueError("partition_key cannot be longer than 128 characters.")
+
+        if value and value != self.session_id:
```

Comment:
To confirm, we don't want this logic in session_id as well for the inverse relationship?

Reply:
right, it's one-way